src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -39,6 +39,251 @@
+    WAYPOINT_ON_DESPAWN_POINT_SOUTH         = 63,
+    WAYPOINT_ON_TROOPER_DESPAWN             = 1,
+
+    SEAT_HAULER_PLAYER                      = 2,
+    SEAT_HAULER_TROOPER_1                   = 3,
+    SEAT_HAULER_TROOPER_2                   = 4,
+    SEAT_HAULER_TROOPER_3                   = 5,
+    SEAT_HAULER_TROOPER_4                   = 6,
+    SEAT_HAULER_TROOPER_5                   = 7
+};
+
+// Horde Hauler - 44731
+struct npc_silverpine_horde_hauler : public ScriptedAI
+{
+    npc_silverpine_horde_hauler(Creature* creature) : ScriptedAI(creature)
+    {
+        Initialize();
+    }
+
+    void Initialize()
+    {
+        // Note: it needs to be active to be able to travel the map correctly.
+        me->setActive(true);
+    }
+
+    void JustAppeared() override
+    {
+        _events.ScheduleEvent(EVENT_START_PATH, 8s);
+    }
+
+    void JustSummoned(Creature* summon) override
+    {
+        // Note: its summons need to be active to be able to perform their actions correctly.
+        summon->setActive(true);
+    }
+
+    void PassengerBoarded(Unit* passenger, int8 seatId, bool apply) override
+    {
+        if (apply && passenger->IsPlayer())
+        {
+            _playerGUID = passenger->GetGUID();
+
+            if (seatId == SEAT_HAULER_PLAYER)
+            {
+                if (Creature* engineer = me->FindNearestCreature(NPC_HORDE_ENGINEER_HAULER, 15.0f, true))
+                {
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    {
+                        if (engineer->IsAIEnabled())
+                            engineer->AI()->Talk(TALK_HAULER_BOARDED, player);
+                    }
+                }
+            }
+        }
+
+        if (!apply && !passenger->IsPlayer())
+        {
+            uint32 pathId = 0;
+
+            switch (seatId)
+            {
+                case SEAT_HAULER_TROOPER_1:
+                    pathId = PATH_TROOPER_1;
+                    break;
+                case SEAT_HAULER_TROOPER_2:
+                    pathId = PATH_TROOPER_2;
+                    break;
+                case SEAT_HAULER_TROOPER_3:
+                    pathId = PATH_TROOPER_3;
+                    break;
+                case SEAT_HAULER_TROOPER_4:
+                    pathId = PATH_TROOPER_4;
+                    break;
+                case SEAT_HAULER_TROOPER_5:
+                    pathId = PATH_TROOPER_5;
+                    break;
+
+                default:
+                    break;
+            }
+
+            passenger->GetMotionMaster()->MovePath(pathId, false);
+        }
+    }
+
+    void WaypointReached(uint32 waypointId, uint32 pathId) override
+    {
+        if (pathId == PATH_FROM_NORTH_TO_SOUTH)
+        {
+            if (waypointId == WAYPOINT_ON_FORSAKEN_HIGH)
+                _events.ScheduleEvent(EVENT_YELL_ON_FORSAKEN_HIGH, 1s);
+
+            if (waypointId == WAYPOINT_ON_SEPULCHER)
```

Comment:
else if or switch

idea behind this is:
if the first condition is met it does NOT check if the other conditions are met, while if you are using only ifs (not else ifs) its checking every conditions sequentially

Reply:
Feedback applied!